inference-engine/src/vpu/graph_transformer/src/stages/interp.cpp

```diff
@@ -80,8 +98,31 @@ void FrontEnd::parseInterp(const Model& model, const ie::CNNLayerPtr& layer, con
     VPU_THROW_UNLESS(outputs.size() == 1,
                      "Interp stage with name {} must have only 1 output, "
                      "actually provided {}", layer->name, outputs.size());
+    ie::details::CaselessEq<std::string> cmp;
+    const auto coord = layer->GetParamAsString(coordTrans, half_pixel);
+    const auto interpMode = layer->GetParamAsString(modeI, linear);
+    InterpolateCoordTransMode coordinateTransMode = InterpolateCoordTransMode::HalfPixel;
+    InterpolateMode mode = InterpolateMode::Linear;
+
+    if (cmp(coord, asymmetric)) {
+        coordinateTransMode = InterpolateCoordTransMode::Asymmetric;
+    } else if (cmp(coord, half_pixel)) {
+        coordinateTransMode = InterpolateCoordTransMode::HalfPixel;
+    } else if (cmp(coord, pytorch_half_pixel)) {
+        coordinateTransMode = InterpolateCoordTransMode::PytorchHalfPixel;
+    } else if (cmp(coord, tf_half_pixel_for_nn)) {
+        coordinateTransMode = InterpolateCoordTransMode::TfHalfPixelForNn;
+    } else if (cmp(coord, align)) {
+        coordinateTransMode = InterpolateCoordTransMode::AlignCorners;
+    } else {
+        VPU_THROW_EXCEPTION << "Current Interp doesn't support this coordinate transformation mode; layer name = " << layer->name;
+    }
+
+    if (cmp(interpMode, linear_onnx)) {
+        mode = InterpolateMode::LinearOnnx;
```

Comment:
I see here mode could be only linear or ONNX linear. Why it does not handle other modes?

Reply:
Because for now nearest mode is handled by resample stage, linear/linear_onnx - by interp. This separation is done in the Interpolate stage.